src/AutoRest.CSharp/MgmtTest/AutoRest/SourceCodeProject.cs

```diff
@@ -41,9 +42,11 @@ public async Task<Compilation> GetCompilationAsync()
         private static Project CreateSourceCodeProject(string sourceCodePath, string[] sharedSourceFolders)
         {
             var sourceCodeProject = CreateGeneratedCodeProject();
-            var sourceCodeGeneratedDirectory = Path.Join(sourceCodePath, GeneratedCodeWorkspace.GeneratedFolder);
+            DirectoryInfo di = new DirectoryInfo(sourceCodePath);
+            var genFolders = di.EnumerateDirectories(GeneratedCodeWorkspace.GeneratedFolder, SearchOption.AllDirectories).ToList();
```

Comment:
the `ToList` here seems redundant?

Reply:
Delay init/iterate also brings higher troubleshooting/maintain cost and sometimes worse perf if used improperly, so it's better to use it only when there is a good reason.